### PR TITLE
Implement new design of settings page - Closes #1995

### LINKED ISSUE #1995: Implement the new design of settings
NG has provided a new design for the settings page. The design indicates many options which we do not have at the moment. We'll migrate all the functionality we have currently to the new design, and later on when we introduce new futures we can use this design as a guide line to add them to the settings page.

### PR DESCRIPTION
**Pull request by massao**
### What issue have I solved?

#1995 

### How have I implemented/fixed it?

- Created `Select` Component to be possible to reuse in the future;
- Created `OutsideClickHandler` Component, to be easier to get when a click was triggered outside a component;
- Update account action, to properly update common informations when no token is present, used for autoLogout feature;
- Refactored `Checkbox` and `SignInTooltipWrapper` to be more generic;
- Updated Setting component to have the new design;
- Rewrite Setting unit tests to use jest.

### How has this been tested?

Go to settings page, should have the new design and changing values, should work the same as before.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/components/setting/setting.js [hunk 1]
```diff
@@ -1,151 +1,199 @@
...
+                <CheckBox
+                  name={'BTC'}
+                  className={`${styles.checkbox} enableBTC`}
+                  checked={settings.token.list.BTC}
+                  onChange={this.handleTokenToggle}
+                />
+                <div>
+                  <span className={styles.labelName}>{t('BTC token')}</span>
+                  <p>{t('By enabling it, you will be able to manage your BTC tokens inside the application.')}</p>
+                </div>
```
Comment (slaweet) [inline src/components/setting/setting.js]: IMO should be displayed only if `localStorage.getItem('btc');`

File: src/components/toolbox/select/index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    const { options, size, className } = this.props;
+    const { selected, isOpen } = this.state;
+    return (
+      <OutsideClickHandler
+        disabled={!isOpen}
+        onOutsideClick={this.toggleIsOpen}
+        className={`${styles.wrapper} ${className}`}
+      >
+        <label className={styles.inputHolder}>
+          <InputV2
```
Comment (slaweet) [inline src/components/toolbox/select/index.js]: Cool idea to reuse the Input inside the select 👍

File: src/components/setting/setting.js (hunk 1 above)
Comment (massao) [reply src/components/setting/setting.js]: Fixed

File: src/components/setting/setting.js [hunk 3]
```diff
@@ -1,151 +1,201 @@
...
+                <CheckBox
+                  name={'advancedMode'}
+                  className={`${styles.checkbox} advancedMode`}
+                  checked={settings.advancedMode}
+                  onChange={this.handleCheckboxChange}
+                />
+                <div>
+                  <span className={styles.labelName}>{t('Delegate Features')}</span>
+                </div>
+              </label>
```
Comment (slaweet) [inline src/components/setting/setting.js]: Now that https://github.com/LiskHQ/lisk-hub/pull/2080 is merged, 'Delegate Features' should be removed from settings, as it's always displayed.

Comment (massao) [reply src/components/setting/setting.js]: Done

**Pull request merged**